### PR TITLE
Support for ControlLayer V2 and updated key exchange protocol

### PR DESCRIPTION
**Pull request by hpihkala**
- Upgrade ControlLayer implementation to [ControlLayer V2](https://github.com/streamr-dev/streamr-specs/blob/master/PROTOCOL.md)
- Implement message validation according to [new spec](https://github.com/streamr-dev/streamr-specs/blob/master/validation.md)
- Update key exchange streams to conform to [new spec](https://github.com/streamr-dev/streamr-specs/blob/master/validation.md)
- Delete StreamMessage V28 and V29 implementations (those version don't occur anymore)
- Small-to-medium cleanup and refactoring
- Activate all [cross-client test scenarios](https://github.com/streamr-dev/streamr-client-testing) in CI (with encrypted tests only using Java client until the JS client can be added)

### CODE REVIEW

File: src/main/java/com/streamr/client/StreamrClient.java [hunk 1]
```diff
@@ -232,29 +234,30 @@ public void connect() throws ConnectionTimeoutException {
...
                             throw new MalformedMessageException("Cannot handle message with content type: " + message.getContentType());
                         }
                     } catch (Exception e) {
-                        log.warn(e.getMessage());
-                        // we don't notify the error to the originator if the message is unauthenticated.
-                        if (message.getSignature() != null) {
-                            StreamMessage errorMessage = msgCreationUtil.createErrorMessage(message.getPublisherId(), e);
-                            publish(errorMessage); //sending the error to the sender of 'message'
-                        }
+                        log.error(e.getMessage());
```
Comment (timoxley) [inline src/main/java/com/streamr/client/StreamrClient.java:260]: Should it "notify the error to the originator"

File: src/main/java/com/streamr/client/StreamrClient.java [hunk 2]
```diff
@@ -590,4 +601,8 @@ private void sendGroupKeyRequest(String streamId, String publisherId, Date start
         StreamMessage request = msgCreationUtil.createGroupKeyRequest(publisherId, streamId, encryptionUtil.getPublicKeyAsPemString(), start, end);
         publish(request);
     }
+
+    private String newRequestId() {
+        return UUID.randomUUID().toString();
```
Comment (timoxley) [inline src/main/java/com/streamr/client/StreamrClient.java:606]: This works of course, but wondering if adding some human-readable part could help with debugging e.g. an incrementing number so it's easy to see the sequence in which the ids were generated, that's definitely helped me in debugging the JS client.

Comment (hpihkala) [reply src/main/java/com/streamr/client/StreamrClient.java:606]: Hmm, could be!

File: src/main/java/com/streamr/client/StreamrClient.java (hunk 1 above)
Comment (hpihkala) [reply src/main/java/com/streamr/client/StreamrClient.java:260]: According to the updated spec, the error message is a failure response to the `GroupKeyRequest` (only). I'm not sure if it makes sense to send error responses to the responses. The old code was doing it though, and also saw that in the JS client. Maybe needs a bit of discussion what the spec should be.

Comment (hpihkala) [reply src/main/java/com/streamr/client/StreamrClient.java:260]: Update: we discussed this and agreed that let's send `GroupKeyErrorResponse` only as a response to `GroupKeyRequest`, but not as a response to any of the responses. Resolving.

**Comment by hpihkala** [comment]
Turns out the key exchange is broken and needs some fixes. I'm splitting that into a separate PR to keep this from dragging on. Reverted this branch to a checkpoint where it had been reviewed, review was addressed, and tests were passing.

**Pull request merged**